build-script.py

```diff
@@ -661,114 +770,116 @@ def run_xctests(toolchain, build_dir, multiroot_data_file, release, verbose):
     )
 
-    build_group.add_argument(
+    generate_source_code_parser.add_argument(
         "--add-source-locations",
         action="store_true",
         help="Insert ###sourceLocation comments in generated code for "
         "line-directive.",
     )
 
-    build_group.add_argument(
-        "--degyb-only",
-        action="store_true",
-        help="The script only generates swift files from gyb and skips the "
-        "rest of the build",
-    )
-
-    build_group.add_argument(
-        "--run-swiftsyntaxbuilder-generation",
+    generate_source_code_parser.add_argument(
+        "--verify",
         action="store_true",
-        help="The script only run the Swift code generation and skips the "
-        "rest of the build",
-    )
-
-    build_group.add_argument(
-        "--disable-sandbox",
-        action="store_true",
-        help="Disable sandboxes when building with SwiftPM",
-    )
-
-    build_group.add_argument(
-        "--multiroot-data-file",
-        help="Path to an Xcode workspace to create a unified build of "
-        "SwiftSyntax with other projects.",
+        help="Instead of generating files using gyb, verify that the files "
+        "which already exist match the ones that would be generated by "
+        "this script.",
     )
 
-    build_group.add_argument(
-        "--toolchain",
-        required=True,
-        help="The path to the toolchain that shall be used to build " "SwiftSyntax.",
-    )
+    # -------------------------------------------------------------------------
+    build_parser = sub_parsers.add_parser("build")
+    build_parser.set_defaults(func=build_command)    
+    
+    add_default_build_arguments(build_parser)
 
     # -------------------------------------------------------------------------
-    test_group = parser.add_argument_group("Test")
+    test_parser = sub_parsers.add_parser("test")
+    test_parser.set_defaults(func=test_command)
 
-    test_group.add_argument("-t", "--test", action="store_true", help="Run tests")
+    add_default_build_arguments(test_parser)
 
-    test_group.add_argument("--skip-lit-tests", action="store_true",
+    test_parser.add_argument("--skip-lit-tests", action="store_true",
         help="Don't run lit-based tests"
     )
 
-    test_group.add_argument(
+    test_parser.add_argument(
         "--filecheck-exec",
         default=None,
         help="Path to the FileCheck executable that was built as part of the "
         "LLVM repository. If not specified, it will be looked up from "
         "PATH.",
     )
 
-    test_group.add_argument(
-        "--gyb-exec",
-        default=GYB_EXEC,
-        help="Path to the gyb tool (default: %(default)s).",
-    )
+    # -------------------------------------------------------------------------
+    verify_source_code_parser = sub_parsers.add_parser("verify-source-code")
+    verify_source_code_parser.set_defaults(func=verify_source_code_command)    
```

Comment:
AFAICT `verify-source-code` also needs a `gyb-exec` parameter because we’re referencing that from the `verify_source_code_command` function.